src/lib/net/index.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * !TODO: add methods functionality, add extra functionality to the Networking (feel like something is missing), add middleware.
+ */
+
+import { DirectoryClass } from "./classes/directory";
+import { RemoteClass } from "./classes/remote";
+import { RouterClass } from "./classes/router";
+
+/* Types */
+import {
+	DirectoryDeclaration,
+	DirectoryObjectDeclaration,
+	EndpointsDeclaration,
+	RemoteObjectDeclaration,
+	RouterDeclaration,
+} from "./types";
+
+export namespace Network {
+	export namespace Method {
+		/**
+		 * The async methods needs some discussion, also the ClientRequest it should ALWAYS be asynchronous, so it lets handle data when is sent back.
+		 */
+
+		export function ClientRequestAsync<T extends () => unknown>(): void {}
```

Comment:
IT WILL expect data, basing it off promises we could potentially just do some syntax like this:

```ts
Event.ClientRequestAsync("getData").andThen((data: T) => {
    ... data handler for request
});
```

Reply:
Nope, never expect, get eaten, remove it!

Reply:
Oke. 😠